### DISCUSSION TITLE
Eslint rule: enforce parenthesis around arrow function arguments

### DISCUSSION THREAD

**Discussion by samturrell** [netsells/code-standards#456] [Ideas]
Hi all,

I see some inconsistency around around functions, and formatting of the arguments when there's a single value. Would be good to come to an agreement on how these should be formatted.

This would be the rule addition: https://eslint.org/docs/rules/arrow-parens

Examples of the proposed rule:

## Examples of incorrect code for this rule with the default "always" option:

```js
/*eslint arrow-parens: ["error", "always"]*/
/*eslint-env es6*/

a => {};
a => a;
a => {'\n'};
a.then(foo => {});
a.then(foo => a);
a(foo => { if (true) {} });
```

## Examples of correct code for this rule with the default "always" option:

```js
/*eslint arrow-parens: ["error", "always"]*/
/*eslint-env es6*/

() => {};
(a) => {};
(a) => a;
(a) => {'\n'}
a.then((foo) => {});
a.then((foo) => { if (true) {} });
```

**Comment by cannibble** [comment]
I'd prefer if we always had the parenthesis. Then it be the same everywhere, not just when needed.

**Reply by cj11489** [reply]
I actually prefer the other way where parentheses are only included when required, but if general consensus is to always have them, thats fine
